src/D2L.CodeStyle.Analyzers/Immutability/ImmutableDefinitionChecker.cs

```diff
@@ -148,9 +151,18 @@ ImmutabilityContext context
 				return true;
 			}
 
-			var decl = prop.GetDeclarationSyntax<PropertyDeclarationSyntax>();
+			if ( prop.IsImplicitlyDeclared ) {

```

Comment:
The first check in `CheckField` is this. Previously there have been no implicitly-declared properties, but now there are.

Reply:
This is important because otherwise we want to find the declaring syntax but there isn't any.